cmd/setProperties.go

```diff
@@ -0,0 +1,120 @@
+				}
+			} else if raw.fromTo != "" {
+				err := strings.Contains(raw.fromTo, "Trash")
+				if !err {
+					return fmt.Errorf("Invalid destination. Please enter a valid destination, i.e. BlobTrash, FileTrash, BlobFSTrash")
+				}
+			}
+			raw.setMandatoryDefaultsForSetProperties()
+			return nil
+		},
+		Run: func(cmd *cobra.Command, args []string) {
+			glcm.EnableInputWatcher()
+			if cancelFromStdin {
+				glcm.EnableCancelFromStdIn()
+			}
+
+			cooked, err := raw.cook()
+			if err != nil {
+				glcm.Error("failed to parse user input due to error: " + err.Error())
+			}
+
+			cooked.commandString = copyHandlerUtil{}.ConstructCommandStringFromArgs()
+			err = cooked.process()
+
+			// TODO -> what is the purpose of these lines?
+			if err != nil {
+				glcm.Error("failed to perform set-properties command due to error: " + err.Error())
+			}
+
+			//if cooked.dryrunMode {
+			//	glcm.Exit(nil, common.EExitCode.Success())
+			//}
+			//
+			glcm.SurrenderControl()
+		},
+	}
+
+	rootCmd.AddCommand(setPropCmd)
+
+	//TODO setPropCmd.PersistentFlags().StringVar(&raw.dst, "dst", "", "The destination location of the copy operation")
+	setPropCmd.PersistentFlags().StringVar(&raw.logVerbosity, "log-level", "INFO", "Define the log verbosity for the log file. Available levels include: INFO(all requests/responses), WARNING(slow responses), ERROR(only failed requests), and NONE(no output logs). (default 'INFO')")
+	setPropCmd.PersistentFlags().StringVar(&raw.include, "include-pattern", "", "Include only files where the name matches the pattern list. For example: *.jpg;*.pdf;exactName")
+	setPropCmd.PersistentFlags().StringVar(&raw.includePath, "include-path", "", "Include only these paths when setting property. "+
+		"This option does not support wildcard characters (*). Checks relative path prefix. For example: myFolder;myFolder/subDirName/file.pdf")
+	setPropCmd.PersistentFlags().StringVar(&raw.exclude, "exclude-pattern", "", "Exclude files where the name matches the pattern list. For example: *.jpg;*.pdf;exactName")
+	setPropCmd.PersistentFlags().StringVar(&raw.excludePath, "exclude-path", "", "Exclude these paths when removing. "+
+		"This option does not support wildcard characters (*). Checks relative path prefix. For example: myFolder;myFolder/subDirName/file.pdf")
+	setPropCmd.PersistentFlags().StringVar(&raw.listOfFilesToCopy, "list-of-files", "", "Defines the location of text file which has the list of only files to be copied.")
+	setPropCmd.PersistentFlags().StringVar(&raw.blockBlobTier, "block-blob-tier", "None", "Changes the access tier of the blobs to the given tier")
+	setPropCmd.PersistentFlags().StringVar(&raw.pageBlobTier, "page-blob-tier", "None", "Upload page blob to Azure Storage using this blob tier. (default 'None').")
+	setPropCmd.PersistentFlags().BoolVar(&raw.recursive, "recursive", false, "Look into sub-directories recursively when uploading from local file system.")
+}
```

Comment:
We're missing the variable to specify which properties to transfer too. It needs to be calculated here so that we can pass it along.

Reply:
wouldn't that be calculated by the code itself? Why do we want to take the values from the user explicitly? Wouldn't it be simpler if we infer the value of that flag depending on what values the user has provided?

Reply:
added